File: src/IM.Production.Domain/Production/ProductionType.cs
using Epam.ImitationGames.Production.Domain.Base;

namespace Epam.ImitationGames.Production.Domain.Production
{
    /// <summary>
    /// Тип производства.
    /// </summary>
    public class ProductionType : BaseProduction, IVisibleEntity
    {
        public string Key { get; set; }

        public string DisplayName { get; set; }

        public override string ToString() => DisplayName;
    }
}

File: src/IM.Production.UI/src/models/Production/ProductionType.ts
import { BaseProduction } from "./BaseProduction";
import { IVisibleEntity } from "../Base/IVisibleEntity";

export class ProductionType extends BaseProduction implements IVisibleEntity {
    constructor(
        public key: string,
        public displayName: string,
        public id: any
    )
    {
        super(id);
    }
}
